### ISSUE TITLE
*Another* Template Variable Warning

### ISSUE THREAD

**Issue by OzGav** [Koenkk/zigbee2mqtt#8769]
<!--

## What happened
I am seeing thousands of these warnings:

```
Logger: homeassistant.helpers.template
Source: helpers/template.py:1485
First occurred: 9:15:11 AM (4774 occurrences)
Last logged: 9:49:31 PM

Template variable warning: 'mappingproxy object' has no attribute 'last_seen' when rendering 'Some Zigbee devices haven't been seen lately... {% for state in states -%} {%- if (state.attributes.last_seen) and not (state.name | regex_search('linkquality|button_fan|update state'))%} {%- if (as_timestamp(now()) - as_timestamp(state.attributes.last_seen) > (60 * 60 * 8) ) %} {{ ((as_timestamp(now()) - as_timestamp(state.attributes.last_seen)) / (3600)) | round(1) }} hours ago for {{ state.name }} {%- endif -%} {%- endif -%} {%- endfor %}'
Template variable warning: 'dict object' has no attribute 'state_button_fan_high' when rendering '{{ value_json.state_button_fan_high }}'
Template variable warning: 'dict object' has no attribute 'state_button_fan_med' when rendering '{{ value_json.state_button_fan_med }}'
Template variable warning: 'dict object' has no attribute 'state_button_fan_low' when rendering '{{ value_json.state_button_fan_low }}'
Template variable warning: 'dict object' has no attribute 'action' when rendering '{{ value_json.action }}'
```

## What did you expect to happen

Not get thousands of warnings? I did create the converter for these light/fan switches so wondering if I missed something in the config? The model of the wall switch is **Smart fan light switch (NUE-ZBFLB)** which is causing all the state_button_fan warnings.

I think the action warnings are being generated by my **MiJia wireless switch (WXKG01LM)**

For the first one i am not sure why I am getting an error about not being able to find the last_seen attribute when the first thing that is checked is the existance of that attribute to avoid just this issue?

## How to reproduce it (minimal and precise)

Unfortunately I am not sure about this. I don't know what is causing it to happen in the first place. I have reviewed the following similar issues but they don't seem to help
https://github.com/Koenkk/zigbee2mqtt/issues/6987
https://github.com/Koenkk/zigbee2mqtt/issues/7361
https://community.home-assistant.io/t/mqtt-dict-object-has-no-attribute/297330/49

## Debug info
Zigbee2MQTT version: 1.18.1-1
Adapter hardware: CC2531
Adapter firmware version: 20190619

**Comment by danieledwardgeorgehitchcock** [comment]
This might have been broken in this release of Home Assistant:

[2021.4](https://www.home-assistant.io/blog/2021/04/07/release-20214/#breaking-changes)

**Comment by OzGav** [comment]
Maybe so. Either way I am hoping to work out how to resolve this.

**Comment by OzGav** [comment]
Bump

**Comment by mwav3** [comment]
The templates in Home assistant need a "default" condition defined now.  I set my own custom template to clear the error and posted it here for the Hampton bay fan https://github.com/Koenkk/zigbee2mqtt/issues/8617

I'm not sure how to fix the template sent during mqtt discovery by zigbee2mqtt,  otherwise I would submit a pr.

The concern is that starting in 2021.4 this is a warning, but per those release notes, will error in a coming release, possibly as soon as 2021.10

So if this isn't fixed, eventually the template will stop working on some future release.

**Comment by OzGav** [comment]
Thanks. Let me try your templatre and see if that removes my warnings and then we can see if we can work out what has to happen to get a PR done.

**Comment by Koenkk** [comment]
Can you check if this has been fixed in the latest-dev? (edge for hassio)

I see you are still using 1.18.1, probably you are using the old repository; check first point of https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.19.0

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Comment by OzGav** [comment]
Yes no warnings in the logs and everything seems to be working correctly (I am getting a warning about units changing from V to mV for my Xiaomi buttons but that seems correct and understandable so can sort that out myself)

Thanks @Koenkk
